### PR TITLE
Simplify streaming reconnection 

### PR DESCRIPTION
**Pull request by briangregoryholmes**
Please go the the `Preview` tab and select the appropriate sub-template:

* [Applications](?expand=1&template=applications_template.md)
* [Others](?expand=1&template=fallback_template.md)

### CODE REVIEW

File: web-common/src/runtime-client/exponential-backoff-tracker.ts [hunk 1]
```diff
@@ -1,52 +1,35 @@
...
 
   public static createBasicTracker() {
-    return new ExponentialBackoffTracker(5, 1000, 250);
+    return new ExponentialBackoffTracker(5, 1000);
   }
 
-  public async failed(): Promise<boolean> {
-    const lastTime = this.curTime;
-    this.curTime = Date.now();
+  public try = async (fn: () => Promise<void> | void) => {
```
Comment (AdityaHegde) [inline web-common/src/runtime-client/exponential-backoff-tracker.ts:17]: Can we keep the older way to defining this class method? While this is valid javascript it is not consistant with other languages.

File: web-common/src/runtime-client/exponential-backoff-tracker.ts [hunk 2]
```diff
@@ -1,52 +1,35 @@
...
+      this.currentDelay = this.initialDelay;
+    } catch (e) {
+      if (this.curRetries >= this.maxRetries) {
+        throw e;
+      }
 
-    // if retry count has reached max return false to stop the connection
-    if (this.curRetries === this.retries) return false;
+      this.currentDelay = this.initialDelay * 2 ** this.curRetries;
+      await new Promise((resolve) => setTimeout(resolve, this.currentDelay));
```
Comment (AdityaHegde) [inline web-common/src/runtime-client/exponential-backoff-tracker.ts:29]: Lets use `asyncWait`

**Review by AdityaHegde — APPROVED** [review]

**Comment by AdityaHegde** [comment]
Lets not merge this until the release is done. Since this changes the behaviour significantly.

**Comment by nishantmonu51** [comment]
@briangregoryholmes @AdityaHegde : Now the release is out, is it ready to be merged ?

**Pull request merged**